app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioGroup.kt

```diff
@@ -0,0 +1,905 @@
+    Horizontal -> HorizontalRadioGroup(
+      data = data,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is HorizontalWithLabel -> HorizontalRadioGroupWithLabel(
+      data = data,
+      groupLabelText = radioGroupStyle.groupLabelText,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      contentPaddingValues = contentPadding,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is Vertical -> VerticalRadioGroup(
+      data = data,
+      radioGroupStyle = radioGroupStyle,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is VerticalWithGroupLabel -> VerticalRadioGroupWithLabel(
+      data = data,
+      radioGroupStyle = radioGroupStyle,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      contentPaddingValues = contentPadding,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+  }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+private fun HorizontalRadioGroup(
+  data: List<OptionData>,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  FlowRow(
+    modifier,
+    maxItemsInEachRow = 2,
+    overflow = Companion.Visible,
+    verticalArrangement = Arrangement.spacedBy(4.dp, Alignment.CenterVertically),
+  ) {
+    for (i in data) {
+      val itemPadding = if (data.indexOf(i) % 2 != 0) {
+        PaddingValues()
+      } else {
+        PaddingValues(end = 4.dp)
+      }
+      if (data.indexOf(i) % 2 == 0 && data.indexOf(i) == data.lastIndex) {
+        RadioOption(
+          data = i,
+          radioOptionStyle = LeftAligned,
+          radioOptionSize = radioGroupSize.toOptionSize(),
+          groupLockedState = groupLockedState,
+          onOptionClick = {
+            onOptionClick(i.id)
+          },
+          modifier = Modifier
+            .weight(1f)
+            .width(Min)
+            .padding(itemPadding),
+          // so with this implementation the downside is
+          // that both types of horizontal groups are not good for optionText longer than 1 word.
+        )
+        Spacer(Modifier.weight(1f))
+      } else {
+        RadioOption(
+          data = i,
+          radioOptionStyle = LeftAligned,
+          radioOptionSize = radioGroupSize.toOptionSize(),
+          groupLockedState = groupLockedState,
+          onOptionClick = {
+            onOptionClick(i.id)
+          },
+          modifier = Modifier
+            .weight(1f)
+            .width(Min)
+            .padding(itemPadding),
+        )
+      }
```

Comment:
I think to avoid having to double call this RadioOption twice, you can make use of the [chunked](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/chunked.html) function to chunck the list in groups of two, and inside the lambda, if the list is actually 1 item then add the spacer instead. You can see it being used here https://github.com/HedvigInsurance/android/blob/c92623cf36c37592aee1772f78b974e8b9303ab9/app/core/core-ui/src/main/kotlin/com/hedvig/android/core/ui/grid/HedvigGrid.kt#L67 too.
I think it would make this a bit easier to write and read perhaps